lib/Sema/SemaExpr.cpp

```diff
@@ -11935,7 +11935,20 @@ ExprResult Sema::ActOnCountBoundsExpr(SourceLocation BoundsKWLoc,
                                       CountBoundsExpr::Kind Kind,
                                       Expr *CountExpr,
                                       SourceLocation RParenLoc) {
-  return new (Context) CountBoundsExpr(Kind, CountExpr, BoundsKWLoc, RParenLoc);
+  ExprResult Result = UsualUnaryConversions(CountExpr);
+  if (Result.isInvalid())
+    return ExprError();
+  Expr *PromotedCountExpr = Result.get();
+  QualType ResultType = PromotedCountExpr->getType();
+  if (!ResultType->isIntegerType()) {
+    std::string CountName = (Kind == CountBoundsExpr::Kind::ElementCount)
+      ? "count" : "byte_count";
```

Comment:
>"count" : "byte_count" [](start = 8, length = 22)

Should these be standard constants somewhere rather than direct literals?  I'm assuming they'll get used elsewhere, and there should already be constants for their use as keywords under bounds expressions, right?